File: src/main/java/by/architecture_map/belarus/service/ConstructionService.java
package by.architecture_map.belarus.service;

import by.architecture_map.belarus.dto.ConstructionDTO;
import by.architecture_map.belarus.entity.Construction;

import java.util.List;

public interface ConstructionService {

    Construction create(Construction construction);

    List<Construction> findAll();

    Construction find(int id);

    /**
     * Using for finding Constructions in Elasticsearch
     */
    List<Construction> find(ConstructionDTO constructionDTO);

    Construction update(int id, Construction updatedConstruction);

    Construction patchUpdate(int id, Construction updatedConstruction);

    void delete(int id);

}

File: src/main/kotlin/by/architecture_map/belarus/service/ConstructionService.kt
package by.architecture_map.belarus.service

import by.architecture_map.belarus.dto.ConstructionDTO
import by.architecture_map.belarus.entity.Construction

interface ConstructionService {

    fun create(construction: Construction): Construction
    fun findAll(): List<Construction>
    fun find(id: Int): Construction

    /**
     * Using for finding Constructions in Elasticsearch
     */
    fun find(constructionDTO: ConstructionDTO): List<Construction>

    fun update(id: Int, updatedConstruction: Construction): Construction
    fun patchUpdate(id: Int, updatedConstruction: Construction): Construction
    fun delete(id: Int)
}
